### PR TITLE
Adds read-only mode to Color Picker Property Editor

### LINKED ISSUE #17045: Readonly Color Picker Property Editor UI
### Which Umbraco version are you using? (Please write the *exact* version, example: 10.1.0)

14.2

### Bug summary

Implement the read-only mode of the Color Picker Property Editor UI.

### Specifics

Here is a link to a PR that implements read-only mode for another Property Editor UI.
https://github.com/umbraco/Umbraco.CMS.Backoffice/pull/2220

**Prerequisites**
https://github.com/umbraco/Umbraco.UI/issues/898
https://github.com/umbraco/Umbraco.UI/issues/899
https://github.com/umbraco/Umbraco.UI/issues/900
https://github.com/umbraco/Umbraco.UI/issues/901
https://github.com/umbraco/Umbraco.UI/issues/902

### Steps to reproduce

Add the `readonly` attribute to the element through developer tools

### Expected result / actual result

It is possible to interact and `readonly` the property editor values but not update them.

### PR DESCRIPTION
**Pull request by richarth**
Implemented the read-only mode of the Color Picker Property Editor UI to close https://github.com/umbraco/Umbraco-CMS/issues/17045.

### CODE REVIEW

File: src/packages/core/components/input-color/input-color.element.ts [hunk 1]
```diff
@@ -11,6 +11,15 @@ import type { UUIColorSwatchesEvent } from '@umbraco-cms/backoffice/external/uui
...
 export class UmbInputColorElement extends UUIFormControlMixin(UmbLitElement, '') {
+	/**
+	 * Sets the input to readonly mode, meaning value cannot be changed but still able to read and select its content.
+	 * @type {boolean}
+	 * @attr
+	 * @default false
+	 */
+	@property({ type: Boolean, reflect: true })
+	readonly = false;
+
```
Comment (bjarnef) [inline src/packages/core/components/input-color/input-color.element.ts:22]: Note it also require to set `readonly` attribute on the underlying rendered `<uui-color-swatches>` element, which hasn't be implemented yet: https://github.com/umbraco/Umbraco.UI/pull/932

**Comment by iOvergaard** [comment]
Awaiting the update to UUI 1.11.0, then we can get this updated and included.

**Review by iOvergaard — APPROVED** [review]

File: src/packages/core/components/input-color/input-color.element.ts [hunk 2]
```diff
@@ -48,7 +51,11 @@ export class UmbInputColorElement extends UUIFormControlMixin(UmbLitElement, '')
 		return map(
 			this.swatches,
 			(swatch) => html`
-				<uui-color-swatch label=${swatch.label} value=${swatch.value} .showLabel=${this.showLabels}></uui-color-swatch>
+				<uui-color-swatch
+					?readonly=${this.readonly}
```
Comment (bjarnef) [inline src/packages/core/components/input-color/input-color.element.ts:55]: @iOvergaard does it need to set this as `readonly` on `uui-color-swatches` explicit set `readonly` on `uui-color-swatch`?
https://github.com/umbraco/Umbraco.UI/pull/932/files#diff-8090b44f925ed46b626e45b3d05f30d71d493772a9ddf02effbb81f86c088475R99

Comment (bjarnef) [reply src/packages/core/components/input-color/input-color.element.ts:55]: It doesn't set `disabled` attribute here either ;)

Comment (iOvergaard) [reply src/packages/core/components/input-color/input-color.element.ts:55]: @bjarnef Good point, but it doesn't look like it's hurting anything setting disabled explicitly. Would you like to make a PR with the disabled attribute?

Comment (bjarnef) [reply src/packages/core/components/input-color/input-color.element.ts:55]: I guess `_handleSlotChange` may execute after? If `<uui-color-swatch>` e.g. inside `<uui-color-swatches>` is set to disabled or readonly, but `<uui-colors-swatches>` isn't, I guess the swatch is still disabled/readonly as it only explicit handle `disbled` and `readonly`.

I guess we may need an `else` to remove attribute as well in `<uui-colors-swatches>`?

https://github.com/umbraco/Umbraco.UI/pull/932/files#diff-8090b44f925ed46b626e45b3d05f30d71d493772a9ddf02effbb81f86c088475R91-R101

**Pull request merged**